consensus/consensus_v2.go

```diff
@@ -23,12 +23,16 @@ func (consensus *Consensus) handleMessageUpdate(payload []byte) {
 	if len(payload) == 0 {
 		return
 	}
-	msg := &msg_pb.Message{}
-	if err := protobuf.Unmarshal(payload, msg); err != nil {
+
+	var m msg_pb.Message
+
+	if err := protobuf.Unmarshal(payload, &m); err != nil {
```

Comment:
what's the benefit of doing this?

Reply:
More idiomatic unmarshal and less needless allocation

Reply:
Actually I think the following way has the least allocation.

```
var m *msg_pb.Message
if err := protobuf.Unmarshal(payload, m); err != nil {...}
```

But I cannot agree more on this idiomatic way of writing unmarshal code.

BTW, did some experiment about unmarshaling with pointer or value, the object will always escape to queue. You were definitely right about it - no difference in GC. And thanks for pointing that out in last video chat :)